app/services/expired_license_allower.rb

```diff
@@ -0,0 +1,64 @@
+# Needs the "raw" response error codes, which needs to happen before translation
+class ExpiredLicenseAllower
+  # @param IdentityDocAuth::Response
+  def initialize(response)
+    @response = response
+  end
+
+  # When we allow expired drivers licenses, returns a successful response
+  # @return [IdentityDocAuth::Response]
+  def processed_response
+    if document_expired?
+      if IdentityConfig.store.proofing_allow_expired_license && allowable_expired_license?
+        IdentityDocAuth::Response.new(
+          success: true, # new response with explicit true overrides success
+          errors: {},
+          exception: response.exception,
+          extra: response.extra.merge(
+            document_expired: document_expired?,
+            reproof_at: reproof_at,
+          ),
+          pii_from_doc: response.pii_from_doc,
+        )
+      else
+        response.merge(
+          IdentityDocAuth::Response.new(
+            success: true, # merge uses "&&" to combine success, so this does not override
+            extra: {
+              document_expired: document_expired?,
+              would_have_passed: allowable_expired_license?,
+            },
+          ),
+        )
+      end
+    else
+      response
+    end
+  end
+
+  def document_expired?
+    !!response.errors[:id]&.include?(IdentityDocAuth::Errors::DOCUMENT_EXPIRED_CHECK)
+  end
+
+  def only_error_was_document_expired?
+    response.errors.keys == [:id] &&
+      response.errors[:id] == [IdentityDocAuth::Errors::DOCUMENT_EXPIRED_CHECK]
+  end
+
+  def allowable_expired_license?
+    !!(
+      only_error_was_document_expired? &&
+        response.pii_from_doc[:state_id_expiration] &&
+        (state_id_expiration = Date.parse(response.pii_from_doc[:state_id_expiration])) &&
```

Comment:
Side-note: I spent a fair bit of time looking at this quite puzzled, until discovering that we use [a gem](https://rubygems.org/gems/american_date) to alter date parsing behavior, as e.g. `Date.parse('05/02/2021') >= Date.parse('2021-05-01')` is `false` in normal circumstances due to how the parsing would defer to the international format in the first instance. I'd wonder if this sort of monkey-patching is worth the potential confusion for new contributors, vs. just using `Date.strptime(date_str, '%m/%d/%y')` or setting the source value in a non-ambiguous date format (like we do with the new config dates here).

It's a higher-level question than I'd expect to solve here and now, but I figured the anecdote was worth sharing.

Reply:
I wish we didn't use the american-style date at all, but it's what our gems have used for a long time, so I didn't want to change that :/

I would be a big fan of only using the umambiguous, international-style only

Reply:
Oh one more thought.... the encrypted PII bundles in the DB likely have the American-style dates embedded in them.... we may need to have some form of backward compatibility forever .. I'll file a ticket to track this